common/src/main/java/dev/architectury/event/events/client/ClientParticleFactoryRegisterEvent.java

```diff
@@ -0,0 +1,17 @@
+package dev.architectury.event.events.client;
+
+import dev.architectury.event.Event;
+import dev.architectury.event.EventFactory;
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+@Environment(EnvType.CLIENT)
+public interface ClientParticleFactoryRegisterEvent {
```

Comment:
I would rather have the event subscribe in the forge's ParticleProviderRegistryImpl fixed, we don't generally have these side-only events, only when it's necessary

Reply:
Fair point. I'll look into just fixing that rather than the route I went. Thanks for reviewing anyway.

Reply:
Hi again, here is why it doesn't seem like it will work. I fixed the event subscribe (all it needed was to point to the Mod event bus in the annotation because Forge bus is default). But the order of which the events are fired does not work. As can be seen in the log below, the particles initialize method is called, which is where the ParticleProviderRegistry.register function is added to the CLIENT_SETUP event. However, that event is fired later via FMLClientSetupEvent, after the ParticleFactoryRegistryEvent is fired. I do believe this makes the case for a new event, whether it is my suggestion or another one which handles an event prior to the ParticleFactoryRegistryEvent.

> TestParticles.initialize
onParticleFactoryRegister
[22:06:12] [Render thread/INFO] (Minecraft) Narrator library for x64 successfully loaded
[22:06:12] [Render thread/INFO] (Minecraft) Reloading ResourceManager: Default, Mod Resources
[22:06:12] [modloading-worker-0/WARN] (ForgeConfigSpec) Configuration file C:\Users\jmils\IdeaProjects\architectury-api\testmod-forge\run\config\forge-client.toml is not correct. Correcting
[22:06:12] [modloading-worker-0/WARN] (ForgeConfigSpec) Incorrect key client was corrected from null to its default, SimpleCommentedConfig:{}. 
[22:06:12] [modloading-worker-0/WARN] (ForgeConfigSpec) Incorrect key client.alwaysSetupTerrainOffThread was corrected from null to its default, false. 
[22:06:12] [modloading-worker-0/WARN] (ForgeConfigSpec) Incorrect key client.experimentalForgeLightPipelineEnabled was corrected from null to its default, false. 
[22:06:12] [modloading-worker-0/WARN] (ForgeConfigSpec) Incorrect key client.showLoadWarnings was corrected from null to its default, true. 
[22:06:12] [modloading-worker-0/WARN] (ForgeConfigSpec) Incorrect key client.useCombinedDepthStencilAttachment was corrected from null to its default, false. 
[22:06:12] [modloading-worker-0/WARN] (ForgeConfigSpec) Incorrect key client.forceSystemNanoTime was corrected from null to its default, false. 
[22:06:12] [modloading-worker-0/WARN] (ForgeConfigSpec) Configuration file C:\Users\jmils\IdeaProjects\architectury-api\testmod-forge\run\config\forge-common.toml is not correct. Correcting
[22:06:12] [modloading-worker-0/WARN] (ForgeConfigSpec) Incorrect key general was corrected from null to its default, SimpleCommentedConfig:{}. 
[22:06:12] [modloading-worker-0/WARN] (ForgeConfigSpec) Incorrect key general.defaultWorldType was corrected from null to its default, default. 
[22:06:12] [Render thread/INFO] (Architectury Test) Client started!
[22:06:12] [Worker-Main-23/ERROR] (Minecraft) Using missing texture, unable to load architectury_test:textures/mob_effect/test_effect.png : java.io.FileNotFoundException: architectury_test:textures/mob_effect/test_effect.png
FMLClientSetupEvent
ClientLifecycleEvent.CLIENT_SETUP.register